src/Features/AssignableAsset.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Features;
+
+use Glpi\DBAL\QueryExpression;
+use Session;
+
+trait AssignableAsset
+{
+    // TODO with PHP 8.2, we can use constants for these rights
+    public static int $read_assigned = 256;
```

Comment:
Not (yet) a critic, I just wanted to know why you chose to go with properties instead of constants in `define.php`

Reply:
I was concerned that there could be cases where 256 and 512 are already used for other permissions with asset itemtypes. If they are declared in the trait, it should be possible to change the value as needed per-itemtype. This seems like it would only be an issue with plugins. I don't see any cases in GLPI itself where 256 and 512 are used by other permissions.